src/Features/CSharp/Portable/MoveToResx/CSharpMoveToResxCodeFixProvider.cs

```diff
@@ -0,0 +1,235 @@
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Composition;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Text;
+using System.Collections.Immutable;
+using System.IO;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Microsoft.CodeAnalysis.CSharp.MoveToResx
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(CSharpMoveToResxCodeFixProvider)), Shared]
+    internal class CSharpMoveToResxCodeFixProvider : CodeFixProvider
+    {
+        [ImportingConstructor]
+        public CSharpMoveToResxCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(MoveToResx.DiagnosticId);
+
+        public override FixAllProvider GetFixAllProvider()
+            => WellKnownFixAllProviders.BatchFixer;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics[0];
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root?.FindNode(diagnostic.Location.SourceSpan) as LiteralExpressionSyntax;
+            if (node == null)
+                return;
+
+            var resxFile = FindBestResxFile(context.Document.Project, context.Document.FilePath);
+            if (resxFile != null)
+            {
+                var resxName = Path.GetFileName(resxFile.FilePath);
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        title: $"Move string to .resx resource ({resxName})",
+                        createChangedSolution: c => MoveStringToResxAndReplaceLiteralAsync(context.Document, node, resxFile, c),
+                        equivalenceKey: "MoveStringToResxInMemory"),
+                    diagnostic);
+            }
+            else
+            {
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        title: "No .resx file found. Please create a .resx file in your project or folder.",
+                        createChangedDocument: c => Task.FromResult(context.Document), // No-op
+                        equivalenceKey: "NoResxFileFound"),
+                    diagnostic);
+            }
+        }
+
+        private static TextDocument? FindBestResxFile(Project project, string? documentPath)
+        {
+            var allResx = project.AdditionalDocuments
+                .Where(f => f.FilePath != null && string.Equals(".resx", Path.GetExtension(f.FilePath), StringComparison.OrdinalIgnoreCase))
+                .ToList();
+
+            if (allResx.Count == 0 || documentPath == null)
+                return allResx.FirstOrDefault();
```

Comment:
You should always have a document path.  so this second clause won't ever hit.

fort he first clause, it's odd since you knwo the count is 0, so returning FirstOrDefualt is equivalent to jsut returning null.